sonoff/xsns_29_vl53l1x.ino

```diff
@@ -0,0 +1,112 @@
+/*
+  xsns_29_vl53l1x.ino - VL53L1X
+
+  Copyright (C) 2018  Theo Arends
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifdef USE_I2C
+#ifdef USE_VL53L1X
+/*********************************************************************************************\
+ * VL53L1X
+ *
+ * Source:
+ *
+ * I2C Address: 0x52
+\*********************************************************************************************/
+
+#include <Wire.h>
+#include "VL53L1X.h"
+VL53L1X sensor;
+
+uint8_t vl53l1x_ready = 0;
+
+/********************************************************************************************/
+
+void Vl53l1Detect()
+{
+  if (vl53l1x_ready) {
+    AddLog_P(LOG_LEVEL_DEBUG, PSTR("Vl53l1 is ready"));
+    return;
+  }
+
+  if (!sensor.init()) {
+    AddLog_P(LOG_LEVEL_DEBUG, PSTR("Unable to begin sensor"));
+    snprintf_P(log_data, sizeof(log_data), S_LOG_I2C_FOUND_AT, "VL53L1X", "Sensor not ready");
+    return;
+  }
+
+  AddLog_P(LOG_LEVEL_DEBUG, PSTR("Detected Vl53l1"));
+
+  sensor.setDistanceMode(VL53L1X::Long);
```

Comment:
@arendst could this setting come from an enum from web settings? Could you provide me with guidance here?

Reply:
No webpages for sensors. Takes up too much space. If parameters are needed use the sensor29 command as used on the MHZ19 sensor